t/08_trento.t

```diff
@@ -846,5 +943,96 @@ subtest '[cluster_wait_status]' => sub {
     ok(scalar @calls == 1, 'cluster_wait_status exit after the first qesap_ansible_script_output call as the output match the regexp');
 };
 
+subtest '[podman_wait] ret 0' => sub {
+    my $trento = Test::MockModule->new('trento', no_auto => 1);
+    @calls = ();
+
+    $trento->redefine(script_run => sub { push @calls, $_[0]; });
+    $trento->redefine(script_output => sub {
+            return 'Exited (0) 38 seconds ago,STRUDEL';
+    });
+    $trento->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+
+    my $ret = podman_wait(name => 'STRUDEL', timeout => 1);
+
+    note("\n  C-->  " . join("\n  C-->  ", @calls));
+    ok $ret == 0;
+    ok((any { /wait.*pgrep.*podman/ } @calls), 'Wait podman process to terminate');
+    ok((!any { /podman.*logs/ } @calls), 'podman logs not called if podman exit 0');
+};
+
+subtest '[podman_wait] support for more than one dessert' => sub {
+    my $trento = Test::MockModule->new('trento', no_auto => 1);
+
+    $trento->redefine(script_run => sub { });
+    $trento->redefine(script_output => sub {
+            return 'Exited (0) 38 seconds ago,KRAPFEN';
+    });
+    $trento->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+
+    my $ret = podman_wait(name => 'KRAPFEN', timeout => 42);
+
+    ok $ret == 0;
+};
+
+subtest '[podman_wait] ret 42' => sub {
+    my $trento = Test::MockModule->new('trento', no_auto => 1);
+    @calls = ();
+
+    $trento->redefine(script_run => sub { push @calls, $_[0]; });
+    $trento->redefine(script_output => sub {
+            return 'Exited (42) 38 seconds ago,STRUDEL';
+    });
+    $trento->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+
+    my $ret = podman_wait(name => 'STRUDEL', timeout => 1);
+
+    note("\n  C-->  " . join("\n  C-->  ", @calls));
+    ok $ret == 42;
+    ok((any { /podman.*logs/ } @calls), 'podman logs not called if podman exit 0');
+};
+
+subtest '[podman_wait] timeout' => sub {
+    my $trento = Test::MockModule->new('trento', no_auto => 1);
+    @calls = ();
+
+    $trento->redefine(script_run => sub { push @calls, $_[0]; });
+    $trento->redefine(enter_cmd => sub { push @calls, $_[0]; });
+    $trento->redefine(script_output => sub {
+            return 'Up 38 seconds ago,STRUDEL';
+    });
+    $trento->redefine(podman_exec => sub { my (%args) = @_; push @calls, $args{cmd}; });
+    $trento->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+
+    my $ret = podman_wait(name => 'STRUDEL', timeout => 1);
+
+    note("\n  C-->  " . join("\n  C-->  ", @calls));
+    ok $ret == 1;
+    ok((any { /pkill.*cypress/ } @calls), 'pkill cypress');
+    ok((any { /pkill.*podman/ } @calls), 'pkill cypress');
```

Comment:
The bug about `$args{container_name}` was not detected by this Unit Test as the mock at like 1005 is only taking care of the cmd and not about the name 
`push @calls, $args{cmd};`